source/utils/formatDate.mc

```diff
@@ -0,0 +1,29 @@
+import Toybox.Lang;
+import Toybox.Time;
+import Toybox.Time.Gregorian;
+
+function formatDate(time as Time.Moment) as Array<String> {
+    var englishDays = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
+    var date = Gregorian.info(time, Time.FORMAT_SHORT);
+    var dayOfWeek = englishDays[date.day_of_week - 1];
+    var dayMonth = "";
+    switch (SettingsModule.getValue(SettingsModule.SettingType.DATE_FORMAT)) {
```

Comment:
Getting settings inside utilities causes excessive code cohesion. Also, when calling a utility, the settings are requested and parsed for each frame, which is redundant, it can be optimized by requesting the setting and saving it only when it is needed, please look at [this commit](https://github.com/shbumba/SimplePixels/commit/10695146e30991df85a176e0e6f48b35a05b3258#diff-56676d6adbb568c19d7318b283ca7174b643c7884ef7c1520dbc5da0dd534b26R46). So we need to move the receiving of settings to the visual component level, and the setting itself can be passed as an argument to the function.
Also as a variant functions `formatDateToEng` and `formatDateToMMdd` can be left at the level of visual component as they are not overused. Thus the setting will be saved in the visual component, updated on demand, visual display and formatting will also take place and be located in it, and the `formatDate` function will only deal with date parsing. 

There is also a suggestion to implement `formatDate` in this way:

```monkey-c

import Toybox.Lang;
import Toybox.Time;
import Toybox.Time.Gregorian;

module FromatDate {
    module DisplayDateFormatType {
        enum Enum {
            DDMM = 0,
            MMDD = 1
        }
    }

    typedef FromatDateResult as {
        :month as Lang.Number,
        :day as Lang.Number,
        :monthName as Lang.String,
        :dayName as Lang.String,
    };

    function formatDate(time as Time.Moment, format as Time.DateFormat?) as FromatDateResult {
        var englishDays = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
        var englishMonths = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];

        var timeFormat = format || Time.FORMAT_SHORT;
        var date = Gregorian.info(time, timeFormat);

        var month = date.month;
        var day = date.day;
        var monthName = englishMonths[month - 1];
        var dayOfWeek = englishDays[date.day_of_week - 1];

        return {
            :month => month,
            :day => day,
            :monthName => monthName,
            :dayName => dayOfWeek
        };
    }

    function formatDateByType(time as Time.Moment, type as DisplayDateFormatType.Enum) as String {
        var formatedDate = formatDate(time, Time.FORMAT_SHORT);

        switch (type) {
            case DisplayDateFormatType.DDMM:
                return formatedDate.day.toString() + " " + formatedDate.monthName;
            
            case DisplayDateFormatType.MMDD:
                return formatedDate.month.toString() + "/" + formatedDate.day.toString();
        }
    }
}
```

Reply:
Thank you very much for letting me learn this.I will revise and submit it again next week.